x-pack/plugins/lens/public/xy_visualization/annotations/helpers.tsx

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import moment from 'moment';
+import { layerTypes } from '../../../common';
+import type {
+  XYDataLayerConfig,
+  XYAnnotationLayerConfig,
+  XYLayerConfig,
+} from '../../../common/expressions';
+import type { FramePublicAPI, Visualization } from '../../types';
+import { isHorizontalChart } from '../state_helpers';
+import type { XYState } from '../types';
+import {
+  checkScaleOperation,
+  getAnnotationsLayers,
+  getAxisName,
+  getDataLayers,
+  isAnnotationsLayer,
+} from '../visualization_helpers';
+import { LensIconChartBarAnnotations } from '../../assets/chart_bar_annotations';
+import { generateId } from '../../id_generator';
+import { defaultAnnotationColor } from '../../../../../../src/plugins/event_annotation/public';
+import { defaultAnnotationLabel } from './config_panel';
+
+const MAX_DATE = Number(new Date(8640000000000000));
```

Comment:
just interesting, why we need it? 

<img width="243" alt="image" src="https://user-images.githubusercontent.com/20072247/159677033-565d4c2c-fbcf-4560-a548-9f1e66aef5b2.png">